### PR TITLE
added a compile hook for `hardhat-vyper` plugin

### PR DESCRIPTION
**Pull request by Saw-mon-and-Natalie**
This PR creates a workflow to compile `Vyper` contracts using `hardhat`. It also adds a preprocessing functionality using `C` like pragmas ( `#pragma once`, `#include "subContract.vy"`, ...).

## Summary of additions and changes

- The `hardhat` configuration is seperated in its own config file and also a new `build:vyper` script is added to `package.json`.
- A new `compile` hook is introduced for the `@nomiclabs/hardhat-vyper` plugin to add the preprocessing functionality.

### CODE REVIEW

**Review by 0age — APPROVED** [review]
Awesome! Have you run `yarn audit` after updating dependencies?

**Comment by Saw-mon-and-Natalie** [comment]
> Awesome! Have you run `yarn audit` after updating dependencies?

@0age , yes, there seem to be 2 low severity vulnerabilities related to `ethereum-waffle > ... > web3` which wasn't added by the new dependencies:

[Insecure Credential Storage in web3](https://www.npmjs.com/advisories/1067669)

File: hardhat.config.ts [hunk 1]
```diff
@@ -23,18 +22,6 @@ subtask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS).setAction(
 // You need to export an object to set up your config
 // Go to https://hardhat.org/config/ to learn more
 
-const getPaths = () => {
```
Comment (Thomasrichard1122) [inline hardhat.config.ts:26]: Release data

File: hardhat.config.ts [hunk 2]
```diff
@@ -23,18 +22,6 @@ subtask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS).setAction(
 // You need to export an object to set up your config
 // Go to https://hardhat.org/config/ to learn more
```
Comment (Thomasrichard1122) [inline hardhat.config.ts:23]: Open

File: hardhat.config.ts [hunk 3]
```diff
@@ -2,7 +2,6 @@ import * as dotenv from "dotenv";
 
 import { HardhatUserConfig, subtask } from "hardhat/config";
 import "@nomiclabs/hardhat-waffle";
-import "@nomiclabs/hardhat-vyper";
 import "@typechain/hardhat";
 import "hardhat-gas-reporter";
```
Comment (Thomasrichard1122) [inline hardhat.config.ts:6]: Open

File: hardhat.config.ts [hunk 4]
```diff
@@ -2,7 +2,6 @@ import * as dotenv from "dotenv";
 
 import { HardhatUserConfig, subtask } from "hardhat/config";
```
Comment (Thomasrichard1122) [inline hardhat.config.ts:3]: Exports

Comment (Thomasrichard1122) [inline hardhat.config.ts:3]: Exports

**Review by Thomasrichard1122 — CHANGES_REQUESTED** [review]

**Pull request merged**